compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt

```diff
@@ -495,11 +496,14 @@ internal class ComposeSceneMediator(
 
     fun viewWillLayoutSubviews() {
         val density = container.systemDensity
-        //TODO: Current code updates layout based on rootViewController size.
-        // Maybe we need to rewrite it for SingleLayerComposeScene.
-
-        val boundsInWindow = windowContext.boundsInWindow(container)
-        scene.density = density // TODO: Maybe it is wrong to set density to scene here?
+        val boundsInWindow: IntRect = if (mainComposeScene) {
+            with(density) {
+                container.bounds.useContents { toDpRect().toRect().roundToIntRect() }
```

Comment:
Why `windowContext.boundsInWindow` doesn't work?

Reply:
Added video to PR description

Reply:
We should modify this condition instead

https://github.com/JetBrains/compose-multiplatform-core/blob/af0ecd40521f33005e315b7122308f780ddc3aff/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt#L68

Probably based on `container` hierarhy / window attachement or so

Reply:
resolved